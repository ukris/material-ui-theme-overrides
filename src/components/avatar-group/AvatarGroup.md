Avatar Group Standard Example:
```jsx
  <AvatarGroup
      avatars={[{src:"https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50", name:"Natasha Special"},
      {name:"John Special"},
      {name:"Natasha Special"},
      {name:"Sasha Special"} ,
      {name:"troy smith"}      
      ]}
      size="lg"
      max="3"
      spacing="separate"
  />
```
Avatar Group Spacing Example:
```jsx
  <AvatarGroup
      avatars={[{src:"https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50", name:"Natasha Special"},
      {src:"https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50", name:"Natasha Special"},
      {src:"https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50", name:"Natasha Special"},{src:"https://www.gravatar2.com/avatar/205e460b479e2e5b48aec07710c08d50", name:"Natasha Special"}      
      ]}
      spacing="together"
  />
```