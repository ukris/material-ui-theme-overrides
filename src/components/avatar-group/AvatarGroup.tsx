import React from 'react'
import clsx from 'clsx'

import { WithStyles, withStyles, Theme} from '@material-ui/core/styles'
import { OverridableComponent } from '@material-ui/core/OverridableComponent'
import { SvgIconTypeMap } from '@material-ui/core/SvgIcon'

import Avatar from 'components/avatar'
import { Shape, Size, IAvatar} from '../avatar/types'

const styles = (_theme: Theme):any => ({
    /* Styles applied to the root element. */
    root: {
      display: 'flex',
    },
    /* Styles applied to the avatar elements. */
    avatar: {
      marginLeft: -8,
    },
});

const SPACINGS = {
    sm: -16,
    md: -12,
    separate: 4,
    together: -8,
  };

export interface StyleProps  extends WithStyles<typeof styles> { 
}

export interface AvatarGroupProps extends StyleProps  {
    children?: React.ReactNode
  /**
   * Max avatars to show before +x.
   */
    avatars: IAvatar[]
    Icon?: OverridableComponent<SvgIconTypeMap<{}, "svg">> | null
    max?: number
    border?: boolean
    shape?: Shape
    size?: Size
    style?: object
  /**
   * Spacing between avatars.
   */
    spacing?: 'sm' | 'md' | 'together' | 'separate' | number
    className?: string
    classes: any
    handleClick?: (e:any) => void
}

function AvatarGroup(props: AvatarGroupProps) {
    const {
      children: childrenProp,
      classes,
      className,
      avatars,
      size,
      shape,
      border,
      spacing = 'md',
      max = 5,
      handleClick,
      Icon,
      ...other
    } = props;
    const extraAvatars = avatars.length > max ? avatars.length - max : 0;

    // @ts-ignore
    const marginLeft = spacing && SPACINGS[spacing] !== undefined ? SPACINGS[spacing] : -spacing
    return (
      <div className={clsx(classes.root, className)}  {...other} data-test="table-avatargroup">
        {avatars.slice(0, avatars.length - extraAvatars).map((child:any, index) =>(
            <Avatar
                key={index}
                src={child.src}    
                name={child.name}
                size={size}
                border={border}
                style= {{
                    zIndex: avatars.length - index,
                    marginLeft
                }}
            />)
        )}
        {extraAvatars ? (
            <Avatar
                className={classes.avatar}
                style={{
                    zIndex: 0,
                    marginLeft,
                }}
                size={size}
                extra
                name={`+${extraAvatars}`}
                border={border}
            />
        ) : null}
        {Icon ? (
          <Avatar
            className={classes.avatar}
            style={{
                zIndex: 1,
                marginLeft: '0.05rem',
            }}
            size={size}
            Icon={Icon}
            border={border}
            handleClick={(e)=> handleClick && handleClick(e)}
      />
        ): null}
      </div>
    )
  }

  export default withStyles(styles)(AvatarGroup)
