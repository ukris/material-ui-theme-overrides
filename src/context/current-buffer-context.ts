import { createContext } from 'react'
import { Buffer } from 'types'

interface ICurrentBufferContextProps {
    currentBuffer?: Buffer[]
}

const CurrentBufferContext  = createContext({} as ICurrentBufferContextProps)

export default CurrentBufferContext