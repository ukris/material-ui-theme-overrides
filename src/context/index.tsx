import CurrentBufferContext from './current-buffer-context'
import TableContext from './table-context'
import GlobalContext from './global-context'
import ProjectContext from './project-context'

export { CurrentBufferContext,  GlobalContext, ProjectContext, TableContext }