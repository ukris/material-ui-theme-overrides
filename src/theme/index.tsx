import { themePalette, LIGHT, DARK } from './palette'
export type LIGHTORDARK = 'light' | 'dark'
export { default } from './theme-provider'
export { default as theme } from './theme'
export { default as makePalette,  } from './palette'
export { default as typography } from './typography'
export { default as ThemeContext } from './theme-context'
export { default as ThemeProvider } from './theme-provider'
export { themePalette, LIGHT, DARK }

const Themes = {
  basil: null,
  crane: null,
  material: null,
  materialDark: null,
  pinky: null,
  rally: null,
  reply: null,
}
export type Color = 'primary' | 'success' | 'info' | 'accent' | 'warning' | 'danger' | 'error' | 'default'
export type ColorValue = {
    light: string,
    main: string,
    dark: string,
    contrastText: string
}
export type ColorObj = {
  [key in Color] : ColorValue
}

export type ThemeType = keyof typeof Themes
export type ThemeZoom = {
  theme: ThemeType
  zoom: number
}
export const AVAILABLE_THEMES = Object.keys(Themes) as ThemeType[]
export const unit = 'rem'
export const tablePadding = [1,0]
export const FONT_BASIS = 2