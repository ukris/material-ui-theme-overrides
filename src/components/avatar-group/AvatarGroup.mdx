---
name: AvatarGroup
route: /avatar-group
---

# AvatarGroup

AvatarGroup

import AvatarGroup from "./AvatarGroup";
import {Playground} from 'docz'

### AvatarGroup

<Playground>
  <AvatarGroup
      avatars={[{src:"https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50", name:"Natasha Special"},
      {name:"John Special"},
      {name:"Natasha Special"},
      {name:"Sasha Special"} ,
      {name:"troy smith"}      
      ]}
      size="lg"
      max="3"
      spacing="separate"
  />
  <AvatarGroup
      avatars={[{src:"https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50", name:"Natasha Special"},
      {src:"https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50", name:"Natasha Special"},
      {src:"https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50", name:"Natasha Special"},{src:"https://www.gravatar2.com/avatar/205e460b479e2e5b48aec07710c08d50", name:"Natasha Special"}      
      ]}
      spacing="together"
  />
</Playground>