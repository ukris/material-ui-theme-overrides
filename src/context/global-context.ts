import { ThemeZoom }     from 'theme'
import { Buffer, FontSize } from 'types'
import  defaults, { Settings } from '../settings'
import { Cell, CellValue, Column } from 'apps/table/types'

const FULL_SCREEN = 'full_screen'
const THEME_ZOOM = 'theme_zoom'
const SETTINGS = 'settings'
const CURRENT_BUFFER = 'buffer'
const BUFFERS = 'buffers'
class LocalStorageMock {
    store:any={}
    constructor() {
      this.store = {};
    }
  
    clear() {
      this.store = {};
    }
  
    getItem(key:string) {
      return this.store[key] || null;
    }
  
    setItem(key:string, value:string) {
      this.store[key] = value.toString();
    }
  
    removeItem(key:string) {
      delete this.store[key];
    }
  };
  if (!global.localStorage) {
      // @ts-ignore
      global.localStorage = new LocalStorageMock()
  }
interface IGlobalContext {
    buffers: Buffer[]
    nextDomainId: number
    nextProjectId: number
    nextBoardId: number
    nextLaneId: number
    settings: Settings
    fullScreen: boolean
    palette?: any
    themeZoom: ThemeZoom,
    fontSize: FontSize,
    setThemeZoom: (themeZoom: ThemeZoom) => void
    setSettings : (settings: Settings) => void
    setFullScreen: (isFullScreen: boolean) => void
    focusedCell?: Cell | null
    currentBuffer?: Buffer
    domainId?: number
    projectId?: number
    boardId?: number
    laneBy?: number
    laneId?: string
    rows: any[]
    columns: Column[]
    anchorEl?: any
    setCurrentBuffer: (buffers: Buffer) => void
    setBuffers: (buffers: Buffer[]) => void
    setDomainId:(domainId: number) => void
    setProjectId:(projectId: number) => void
    setBoardId:(boardId: number) => void
    setLaneId:(laneId: string) => void
    setLaneBy:(labeBy: number) => void
    setRows: (rows:any[]) => void
    setColumns: (columns: Column[]) => void
    getColumn: (idx: number) => Column
    updateCell: (cell: Cell, newVal: CellValue) => void
    setFocusedCell:  (cell: Cell|null) =>void
    setAnchorEl: (el:any) => void
    theme: any
    setTheme: (theme:any)  => void
    /*

    setDomainId: (domainId: number) => void
    setProjectId: (projectId: number) => void
    setBoardId: (boardId: number) => void
    setLaneBy?: (laneBy?: string) => void // split by/ group by colId
    setLaneId: (laneId: string) => void // example  laneby status, laneId = 'FINISHED, 'INPROGRESS' etc 
    setRowIndex: (idx: number) => void
    setColIndex: (idx: number) => void
    setRowId: (id: string) => void
    setColId: (id: string) => void
    setRange: (range: number[]) => void
    */
}

const fullScreen = !!(localStorage && localStorage.getItem(FULL_SCREEN))
const tZoom = localStorage && localStorage.getItem(THEME_ZOOM)
const themeZoom:ThemeZoom = tZoom ? JSON.parse(tZoom) as ThemeZoom : { zoom: 1, theme: 'material'}
const defaultS = localStorage && localStorage.getItem(SETTINGS)
const defaultSettings: Settings = defaultS ? JSON.parse(defaultS) : defaults
let lBuffer = localStorage && localStorage.getItem(CURRENT_BUFFER)
const currentBuffer:Buffer = lBuffer ? JSON.parse(lBuffer) : []
lBuffer = localStorage && localStorage.getItem(BUFFERS)
const buffers:Buffer[] = lBuffer ? JSON.parse(lBuffer) : []

export const GlobalContext: IGlobalContext = {
    buffers,
    currentBuffer,
    nextDomainId: 2,
    nextProjectId: 2,
    nextBoardId: 2,
    nextLaneId: 2,
    fullScreen,
    themeZoom,
    settings: defaultSettings,
    theme: null,
    fontSize: {
        xs: '0.5rem',
        sm: '0.75rem',
        md: '1rem',
        lg: '1.25rem',
        xl: '1.5rem'
    },
    columns: [],
    rows: [],
    setFocusedCell: (cell: Cell|null)  => {
        GlobalContext.focusedCell = cell;
    },
    setThemeZoom: (themeZoom:ThemeZoom) => {
        (async() => localStorage && localStorage.setItem(THEME_ZOOM, 
            JSON.stringify(themeZoom)))()
        GlobalContext.themeZoom = themeZoom
    },
    setSettings: (settings: Settings) => {
        (async() => localStorage  && localStorage.setItem(SETTINGS, 
            JSON.stringify(settings)))()
        GlobalContext.settings = settings
    },
    setCurrentBuffer: (buffer: Buffer) => {
        (async() => localStorage  && localStorage.setItem(CURRENT_BUFFER, 
            JSON.stringify(buffer)))()
        GlobalContext.currentBuffer = buffer
    },
    setBuffers: (buffers: Buffer[]) => {
        (async() => localStorage  && localStorage.setItem(BUFFERS, 
            JSON.stringify(buffers)))()
        GlobalContext.buffers = buffers
    },
    setFullScreen: (screen: boolean) => {
        (async() => localStorage  && localStorage.setItem(FULL_SCREEN, screen.toString()))()
        GlobalContext.fullScreen = screen
    },
    setDomainId: (domainId: number) => {
        GlobalContext.domainId = domainId
    },
    setProjectId: (projectId: number) => {
        GlobalContext.projectId = projectId
    },
    setBoardId: (boardId: number) => {
        GlobalContext.boardId = boardId
    },
    setLaneId:(laneId: string) => {
        GlobalContext.laneId = laneId
    },
    setLaneBy:(laneBy: number) => {
        GlobalContext.laneBy = laneBy
    },
    setRows: (rows: any[]) => {
        GlobalContext.rows = rows
    },
    setColumns: (columns: Column[]) => {
        GlobalContext.columns = columns
    },
    getColumn: (idx:number) => GlobalContext.columns[idx],
    updateCell: (cell: Cell, newVal: CellValue) => {
        const { rowIndex, colIndex } = cell
        GlobalContext.rows[rowIndex][colIndex] = newVal
    },
    setAnchorEl: (el:any) => {
        GlobalContext.anchorEl = el
    },
    setTheme: (theme: any) => {
        GlobalContext.theme = theme
    }
}

export default GlobalContext