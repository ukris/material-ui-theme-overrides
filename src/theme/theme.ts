import { createMuiTheme, responsiveFontSizes} from '@material-ui/core'
import PaletteOptions from '@material-ui/core/styles/createMuiTheme';
import { GlobalContext } from 'context/global-context'
import { FONT_BASIS, makePalette, ThemeZoom, unit, tablePadding } from './index'
import { roundNum } from 'helpers'
// https://www.sipios.com/blog-tech/how-to-use-styled-components-with-material-ui-in-a-react-app
import typography from './typography'

function createTheme(variant: ThemeZoom):any {
  let palette:any = makePalette(variant)
  const { zoom } = variant
  const { colors } = palette
  delete palette.colors
  GlobalContext.palette = colors
  GlobalContext.palette.type = palette.type
  GlobalContext.palette.contrastText = palette.contrastText
  palette = {...palette, ...colors}
  GlobalContext.setThemeZoom(variant)
  const theme = createMuiTheme({
    zoomFontSize: (factor:number):string => `${roundNum(0.30 * factor * zoom)}${unit}`,
    zoomSpacing: (factor: number) => spacing(roundNum(factor * zoom)),
    fontSize: GlobalContext.fontSize,
    spacing: factor => `${roundNum(0.60 * factor)}${unit}`,
    lineHeight: GlobalContext.fontSize
  });
  const { spacing, zoomSpacing, zoomFontSize } = theme
  const fontSize =  {
      xs: zoomFontSize(FONT_BASIS),
      sm: zoomFontSize(FONT_BASIS+1),
      md: zoomFontSize(FONT_BASIS+2),
      lg: zoomFontSize(FONT_BASIS+3),
      xl: zoomFontSize(FONT_BASIS+4)
  }
  const lineHeight =  {
    xs: zoomFontSize(FONT_BASIS+0.5),
    sm: zoomFontSize(FONT_BASIS+1.5),
    md: zoomFontSize(FONT_BASIS+2.5),
    lg: zoomFontSize(FONT_BASIS+3.5),
    xl: zoomFontSize(FONT_BASIS+4.5)
  }
  let padding:string = tablePadding[0].toString()
  if (tablePadding[0]) {
    padding+= unit
  }
  padding+= ' '
  padding+= tablePadding[1].toString()
  if (tablePadding[1]) {
    padding+= unit
  }
  GlobalContext.fontSize = fontSize
  const modTheme:any = createMuiTheme(
    /**
     * @see https://material-ui.com/customization/themes/#theme-configuration-variables
     * https://material-ui.com/customization/typography/
     */
    {
      palette,
      zoomFontSize,
      typography,
      spacing,
      zoomSpacing,
      fontSize,
      lineHeight,
      props: {
        MuiButtonBase: {
          disableRipple: true
        },
        MuiButton: {
            variant: 'contained',
            disableElevation: true,
        },
        MuiTextField: {
            variant: 'outlined'
        }
      },
      zIndex: {
        appBar: 1200,
        drawer: 1100,
      },
      overrides: {
        MuiDrawer: {
          paper: {
            background: '#18202c',
          },
        },
        MuiPaper: {
          // Name of the component ⚛️ / style sheet
          elevation0: {
            // Name of the rule
            boxShadow: '0 0 14px 0 rgba(53,64,82,.05)', // Some CSS
          },
        },
        MuiTableCell: {
          root: { 
            borderBottom: '1px solid rgba(224, 224, 224, .5)', 
            lineHeight: lineHeight.md,
            fontSize: fontSize.md,
            backgroundColor: theme.palette.background.default,
          },
        },
        MuiAutocomplete: {
          root: {
            backgroundColor: theme.palette.background.default,
            color: theme.palette.getContrastText(theme.palette.background.default)
          }
        },
        MuiButton: {
          root: {
            color: theme.palette.primary.contrastText,
          }
        },
        MuiTextField: {
          root: {
            variant: 'outlined'
          }
        }
    }}
  )
  //  return responsiveFontSizes(modTheme)
  GlobalContext.theme = theme
  return modTheme
}

export default createTheme