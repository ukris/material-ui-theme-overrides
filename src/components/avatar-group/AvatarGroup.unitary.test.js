import React from 'react'
import { GlobalContext } from 'context'
import renderer from 'react-test-renderer'
import { mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import AvatarGroup from './AvatarGroup'
import { ThemeContext, ThemeProvider } from 'theme'

configure({ adapter: new Adapter() })

const themeZoom = GlobalContext.themeZoom;
const setThemeZoom = () => true

describe('AvatarGroup Component | No max prop', () => {
    const component = (
      <ThemeContext.Provider value={{ themeZoom, setThemeZoom }}>
        <ThemeProvider>
            <AvatarGroup
                avatars={[
                {src:"https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50", name:"Natasha Special"},
                {name:"John Special"},
                {name:"Natasha Special"},
                {name:"Sasha Special"} ,
                {name:"troy smith"}      
                ]}
            />
        </ThemeProvider>
      </ThemeContext.Provider>
    )

    const wrapper = mount(component)

    it('renders component correctly', () => {
        const avatarGroupContainer = wrapper.find('[data-test="table-avatargroup"]')

        expect(avatarGroupContainer.length).toEqual(1)
    });

    it('renders correct amounts of Avatars', () => {
        const imgs = wrapper.find('[data-test="table-avatar"]')

        expect(imgs.length).toEqual(5)
    });

    it('matches snapshot', () => {
        const tree = renderer.create(component,).toJSON()

        expect(tree).toMatchSnapshot()
    });
})

describe('AvatarGroup Component | With max prop', () => {
    const component = (
      <ThemeContext.Provider value={{ themeZoom, setThemeZoom }}>
        <ThemeProvider>
            <AvatarGroup
                avatars={[
                {src:"https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50", name:"Natasha Special"},
                {name:"John Special"},
                {name:"Natasha Special"},
                {name:"Sasha Special"} ,
                {name:"troy smith"}      
                ]}
                max="2"
            />
        </ThemeProvider>
      </ThemeContext.Provider>
    )

    const wrapper = mount(component)

    it('renders component correctly', () => {
        const avatarGroupContainer = wrapper.find('[data-test="table-avatargroup"]')

        expect(avatarGroupContainer.length).toEqual(1)
    });

    it('renders correct amounts of Avatars', () => {
        const imgs = wrapper.find('[data-test="table-avatar"]')

        expect(imgs.length).toEqual(3)
    });

    it('matches snapshot', () => {
        const tree = renderer.create(component,).toJSON()

        expect(tree).toMatchSnapshot()
    });
})