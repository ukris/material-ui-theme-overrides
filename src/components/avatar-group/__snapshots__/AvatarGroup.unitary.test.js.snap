// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGroup Component | No max prop matches snapshot 1`] = `
Array [
  " ",
  <div
    className="AvatarGroup-root-33"
    data-test="table-avatargroup"
  >
    <span
      className="Avatar-root-35 Avatar-noClass-39 Avatar-circle-37 Avatar-md-46"
      data-test="table-avatar"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#707264",
          "color": "#fff",
          "marginLeft": -12,
          "zIndex": 5,
        }
      }
    >
      NS
    </span>
    <span
      className="Avatar-root-35 Avatar-noClass-39 Avatar-circle-37 Avatar-md-46 Avatar-colorDefault-36"
      data-test="table-avatar"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#512c66",
          "color": "#fff",
          "marginLeft": -12,
          "zIndex": 4,
        }
      }
    >
      JS
    </span>
    <span
      className="Avatar-root-35 Avatar-noClass-39 Avatar-circle-37 Avatar-md-46 Avatar-colorDefault-36"
      data-test="table-avatar"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#707264",
          "color": "#fff",
          "marginLeft": -12,
          "zIndex": 3,
        }
      }
    >
      NS
    </span>
    <span
      className="Avatar-root-35 Avatar-noClass-39 Avatar-circle-37 Avatar-md-46 Avatar-colorDefault-36"
      data-test="table-avatar"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#957566",
          "color": "#fff",
          "marginLeft": -12,
          "zIndex": 2,
        }
      }
    >
      SS
    </span>
    <span
      className="Avatar-root-35 Avatar-noClass-39 Avatar-circle-37 Avatar-md-46 Avatar-colorDefault-36"
      data-test="table-avatar"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "0",
          "color": "#fff",
          "marginLeft": -12,
          "zIndex": 1,
        }
      }
    >
      ts
    </span>
  </div>,
]
`;

exports[`AvatarGroup Component | With max prop matches snapshot 1`] = `
Array [
  " ",
  <div
    className="AvatarGroup-root-49"
    data-test="table-avatargroup"
  >
    <span
      className="Avatar-root-51 Avatar-noClass-55 Avatar-circle-53 Avatar-md-62"
      data-test="table-avatar"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#707264",
          "color": "#fff",
          "marginLeft": -12,
          "zIndex": 5,
        }
      }
    >
      NS
    </span>
    <span
      className="Avatar-root-51 Avatar-noClass-55 Avatar-circle-53 Avatar-md-62 Avatar-colorDefault-52"
      data-test="table-avatar"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#512c66",
          "color": "#fff",
          "marginLeft": -12,
          "zIndex": 4,
        }
      }
    >
      JS
    </span>
    <span
      className="Avatar-root-51 Avatar-noClass-55 Avatar-circle-53 Avatar-md-62 Avatar-colorDefault-52 AvatarGroup-avatar-50"
      data-test="table-avatar"
      onClick={[Function]}
      style={
        Object {
          "marginLeft": -12,
          "zIndex": 0,
        }
      }
    >
      +3
    </span>
  </div>,
]
`;
